### DISCUSSION TITLE
Automated Testing

### DISCUSSION THREAD

**Discussion by SeanAlling-DojoFive** [sierrafoxtrot/srecord#45] [Ideas]
Currently  shell scrips make up a bulk of `ctest` back end for testing SRecord. What are opinions on using a different tool instead of shell, for example Python. 

With shell scripts, additional steps are required to setup a test environment on Windows. If Python could be used in place of shell scripts, then testing could be performed easily on platforms supported by Python.

**Comment by jtxa** [comment]
Given the age of SRecord, shell was the best answer for portability.

With the usage of CMake, I would say native CMake tests are a nearby alternative.
For the use case "calling an executable" and a little bit of file content check, it should be sufficient, simple and readable.

Python would be an additional dependency. So this must be actively decided.
Personally I don't think we need the power of Python for SRecord related tests.

That being said: If the documentation is probably managed as `.rst` files in future, then maybe the documentation is generated with Sphinx, which is Python based.

**Reply by SeanAlling-DojoFive** [reply]
From experiments between different file formats, `.rst` was able to produce results closest to what is currently generated through `groff`. 

If we use `.rst`, then it would make sense to use a Sphinx which by proxy Python. 

I guess the larger question is, how should testing be preformed? black box as is done now, separate executables per module executing specific features? Combination of both?

**Comment by jtxa** [comment]
Another topic would be naming: the numbers were generated by Aegis, which enforced it for every change. But today I would prefer some readable name.
I suggest something like `test/format/intel.cmake` and `test/filter/crop.cmake` on which the concrete test focuses. (Knowing that every test also uses other flags.)
